### PR TITLE
Add skeleton for Azure Bakery (Only deploys a Resource Group)

### PR DESCRIPTION
**Pull request by ciaranevans**
## What I am changing

TL;DR: This PR creates the shell for the `pangeo-forge-azure-bakery` repository. It provides instructions on setting up the Azure credentials etc needed for a deployment, provides a very small deployable of a named & tagged resource group, and it provides a very basic Github Actions workflow for linting

## How I did it

* Add `terraform/`
    * `main.tf` currently just sets up the `azurerm` provider, the tags, and the named resource group
    * `variables.tf` declares the variables needed during deployment (Populated from `deployment.tfvars.json`)
* Add README
    * Highlights requirements for deploying the Bakery (so far)
    * Provides instructions on authenticating with Azure CLI and then creating the Service Principal for deployment
    * Provides instructions on creating `deployment.tfvars.json`
    * Gives an overview of the provided `Makefile` and the deployment process
* Add `.github/`
    * Adds PR Template
    * Adds simple Github Actions workflow that installs Terraform, initialises it then lints.
    
## How you can test it

Firstly I'd recommend checking out this branch, installing the requirements, and following the Azure setup instructions. From there, you can then run:

```bash
$ make plan
```

To see what you would deploy, you can then run:

```bash
$ make apply
```

To deploy the current stack, which is just a named resource group: `<identifier>-resource-group`, you can then destroy it with:

```bash
$ make destroy
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,216 @@
...
+
+```bash
+$ az role assignment create --assignee "<objectId>" --role "Storage Blob Data Contributor"
+$ az role assignment create --assignee "<objectId>" --role "User Access Administrator"
+$ az role assignment create --assignee "<objectId>" --role  "Azure Kubernetes Service Cluster User Role"
+```
+
+You should now be setup with the correct permissions to deploy the infrastructure onto Azure. Further reading on Azure Service Principals can be found [here](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest).
+
+### `deployment.tfvars.json` file
```
Comment (TomAugspurger) [inline README.md:153]: Is this playing the same role as https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_client_secret? If so, I have a slight preference to use the environment there for authenticating the service principal.

File: terraform/.terraform.lock.hcl [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# This file is maintained automatically by "terraform init".
```
Comment (TomAugspurger) [inline terraform/.terraform.lock.hcl:1]: Should this lock file be gitignored?

File: README.md (hunk 1 above)
Comment (ciaranevans) [reply README.md:153]: So the only real reason I'm passing it in as a variable is that https://learn.hashicorp.com/tutorials/terraform/aks#set-up-and-initialize-your-terraform-workspace the K8s cluster takes it as a parameter... The `az login` handles the actual authenticating for terraform to deploy on your behalf afaik. You just need to pass it to the cluster.

If I've understood correctly.

File: terraform/.terraform.lock.hcl (hunk 2 above)
Comment (ciaranevans) [reply terraform/.terraform.lock.hcl:1]: This is locking the provider versions, it's not locking state etc - I'd argue it should remain so that my deployment runs with the same provider versions yours or a CI build might

Comment (TomAugspurger) [reply terraform/.terraform.lock.hcl:1]: Sounds good, thanks.

File: README.md (hunk 1 above)
Comment (TomAugspurger) [reply README.md:153]: Ah I misunderstood, this is for AKS.

I've only ever used "SystemAssigned" / managed identity: 


```hcl
  identity {
    type = "SystemAssigned"
  }
```

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster

https://docs.microsoft.com/en-us/azure/aks/use-managed-identity

That seemed a bit simpler, and I don't think I've had any issues.

Comment (ciaranevans) [reply README.md:153]: Ahh. Okay. I'll remove the AppId etc then from the vars file.

Comment (ciaranevans) [reply README.md:153]: Done now @TomAugspurger cheers

**Review by TomAugspurger — APPROVED** [review]
